dev/codehost_testing/repo.go

```
@@ -0,0 +1,126 @@
+package codehost_testing
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/go-github/v53/github"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// Repo represents a GitHub repository in the scenario
+type Repo struct {
+	// s is the GithubScenario instance this repo is part of. Actions it creates will be added to this scenario
+	s *GithubScenario
+	// team is the team this repo belongs to
+	team *Team
+	// org is the Org that owns this repo
+	org *Org
+	// name is the name of the repo
+	name string
+}
+
+// Get returns the corresponding GitHub Repository object that was created by the `CreateOrg`
+//
+// This method will only return a Repository if the Scenario that created it has been applied otherwise
+// it will panic.
+func (r *Repo) Get(ctx context.Context) (*github.Repository, error) {
+	if r.s.IsApplied() {
+		return r.get(ctx)
+	}
+	panic("cannot retrieve repo before scenario is applied")
```

`t.Fatal` ?

---

Ultimately decided that to rather return an error. I think ergonomically it can be frustating if a library/package makes your stuff quit.